packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,285 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+import 'package:yaml/yaml.dart';
+
+import '../../artifacts.dart';
+import '../../base/file_system.dart';
+import '../../base/io.dart';
+import '../../base/logger.dart';
+import '../../globals.dart' as globals;
+import '../build_system.dart';
+import '../depfile.dart';
+
+/// Run the localizations generation script with the configuration [options].
+Future<Depfile> generateLocalizations({
+  @required LocalizationOptions options,
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Logger logger,
+  @required Directory projectDir,
+  @required String dartBinaryPath,
+}) async {
+  // Setup project inputs and outputs. This currently makes some
+  // guess about the output and does not include all files.
+  final Directory inputArb = fileSystem.directory(
+    options.arbDirectory ?? projectDir
+      .childDirectory('lib')
+      .childFile('l10n').uri,
+  );
+
+  File outputLocalzations;
+  if (options.outputLocalizationsFile != null) {
+    outputLocalzations = fileSystem
+      .file(inputArb.uri.resolveUri(options.outputLocalizationsFile));
+  } else {
+    outputLocalzations = projectDir
+    .childDirectory('lib')
```

Comment:
Should we be spamming the top-level lib by default? Should we prep for the generated package by putting this by default under something like `lib/src/flutter_gen` ?

Reply:
for better or worse its the gen_l10n tool default

Reply:
Can the tool pass a `--output-localization-file` flag to `gen_l10n` if there isn't one specified in the yaml file?